### PR TITLE
Space-Traveler's-Hub

### PR DESCRIPTION
**Pull request by bhushan354**
Hello Reviewer ,
In this Project me(@bhushan354 ) and my coding partner @Izebiafe have made a Space travelers website for space Enthusiasts,
Below are the list of things that we have used In our code :

- Used React components
- Used React props
- Used Redux store
- Used spacex'x Api to fetch live data
- Used React Router
- Connected React and Redux
- Handled events in a React app
- Wrote unit tests with React Testing Library
- Used styles in a React app
- Used React hooks
- Used Linters
- Applied React best practices and language style guides in code
- Used store, actions and reducers in React
- Performed a code review for a team member

Thank you ,
Bhushan & izabief .

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+              <li>
+                <Link to="/">Rockets</Link>
+              </li>
+              <li>
+                <Link to="/missions">Missions</Link>
+              </li>
+              <li>
+                <Link to="/MyProfile">MyProfile</Link>
+              </li>
+            </div>
```
Comment (codepantha) [inline src/App.js:29]: - [ ] Please use `NavLink` for navigation and ensure that only the active page is indicated with an underline as per requirements 👍
### Design Guideline
image
### Your Navigation
image

File: src/components/Rockets.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          onClick={() => {
+            if (isReserved) {
+              dispatch(removeReserved(id));
+            } else {
+              dispatch(addReserved(id));
+            }
+          }}
+        >
+          Reserve Rocket
+        </button>
```
Comment (codepantha) [inline src/components/Rockets.js:47]: - [ ] Please style the "Reserve Rocket" and "Cancel Reservation" button according to the design guideline 🙏 
### Design Guideline
image

### Your implementation
image

File: src/components/Rockets.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  isReserved: PropTypes.bool.isRequired,
+  dispatch: PropTypes.func.isRequired,
+};
+
+const Rockets = () => {
+  const dispatch = useDispatch();
+  const { rocketsData, isLoading, hasError } = useSelector((store) => store.rocketsData);
+  useEffect(() => {
+    dispatch(getRocketsApi());
+  }, [dispatch]);
```
Comment (codepantha) [inline src/components/Rockets.js:67]: - [ ] Rockets data is re-fetched anytime the `Rocket` component is rendered and this leads to a loss of reserved rockets data. Please ensure that actions are dispatched only once 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ tests are passing 📌 
✔️ Used GitFlow 📌 
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by bhushan354** [comment]
Thank you for the review sir I have Implemented the changes suggested by you.

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @bhushan354 and @Izebiafe , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested changes implemented
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark:  Good commit messages

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- NA

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**